EMBRACE_Client/AssessmentActivityViewController.m

```diff
@@ -0,0 +1,349 @@
+        AnswerOption2EnglishAudio = [currAssessmentActivityStep Answer2Audio];
+        [AnswerOptions addObject:[currAssessmentActivityStep Answer3]];
+        AnswerOption3EnglishAudio = [currAssessmentActivityStep Answer3Audio];
+        [AnswerOptions addObject:[currAssessmentActivityStep Answer4]];
+        AnswerOption4EnglishAudio = [currAssessmentActivityStep Answer4Audio];
+        
+        playAudioFileClass = [[PlayAudioFile alloc]init];
+        
+        //log loading assessment activity
+        [[ServerCommunicationController sharedManager] logComputerAssessmentDisplayStep:Question :AnswerOptions :@"Next" :@"Start Assessment" :BookTitle :ChapterTitle : @"1"];
+        
+    }
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    // Do any additional setup after loading the view from its nib.
+    nextButton.hidden =true;
+    
+    ChapterTitleLabel.text = ChapterTitle;
+}
+
+- (void)didReceiveMemoryWarning
+{
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+//not being used currently may delete
+-(void)loadNextAssessmentActivityQuestion{
+    
+    currentAssessmentActivityStep++;
+    
+    if(currentAssessmentActivityStep<totalAssessmentActivitySteps)
+    {
+        
+        //AnswerList = [[UITableView alloc]init];
+        AnswerSelection[0] = 0;
+        AnswerSelection[1] = 0;
+        AnswerSelection[2] = 0;
+        AnswerSelection[3] = 0;
+        
+        AssessmentActivity* currAssessmentActivityStep = [currentAssessmentActivitySteps objectAtIndex:currentAssessmentActivityStep-1];
+        correctSelection = [currAssessmentActivityStep expectedSelection];
+        Question = [currAssessmentActivityStep QuestionText];
+        questionNum = [currAssessmentActivityStep QuestionNumber];
+        [AnswerOptions removeAllObjects];
+        [AnswerOptions addObject:[currAssessmentActivityStep Answer1]];
+        [AnswerOptions addObject:[currAssessmentActivityStep Answer2]];
+        [AnswerOptions addObject:[currAssessmentActivityStep Answer3]];
+        [AnswerOptions addObject:[currAssessmentActivityStep Answer4]];
+        
+        //log load next assessment activity step
+        [[ServerCommunicationController sharedManager] logComputerAssessmentDisplayStep:Question :AnswerOptions :@"Next" :@"Display Assessment" :BookTitle :ChapterTitle : [NSString stringWithFormat:@"%d", currentAssessmentActivityStep]];
+        
+    }
+    else
+    {
+        //log end of assessment activity
+        
+        [super.navigationController popViewControllerAnimated:YES]; //return to library view
+    }
+}
+
+
+//log user pressed play audio
+-(IBAction)PlayQuestionAudioPressed:(id)sender
+{
+    //[playAudioFileClass playAudioFile:QuestionAudio];
+    [playAudioFileClass textToSpeech:Question];//Question:BookTitle:ChapterTitle:CurrentPage:CurrentSentence:CurrentStep];
+}
+
+-(IBAction)PlayAnswer1AudioPressed:(id)sender
```

Comment:
Since these functions are so similar, is it possible to combine them into a single function?